### PR TITLE
doc: split environment variables from landing page 

### PR DESCRIPTION
**Pull request by coryan**
The landing pages for the reference docs are too cluttered. This does
not fix all the problems, but it is a good start.

Part of the work for #10157

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/11381)

### CODE REVIEW

**Review by devbww — APPROVED** [review]

File: generator/internal/scaffold_generator.cc [hunk 1]
```diff
@@ -676,6 +654,57 @@ can override the default policies.
...
 }
 
+void GenerateDoxygenEnvironmentPage(
+    std::ostream& os, std::map<std::string, std::string> const& variables) {
+  auto constexpr kText = R"""(/*!
+
+@page $library$-env Environment Variables
+
+A number of environment variables can be used to configure the behavior of
+the library.  There are also functions to configure this behavior in code. The
```
[the remark this answers is not in the document]
Comment (coryan) [reply generator/internal/scaffold_generator.cc]: True, fixed here.

File: generator/internal/scaffold_generator.cc [hunk 2]
```diff
@@ -676,6 +654,57 @@ can override the default policies.
...
+environment variables are convenient when troubleshooting problems.
+
+@section $library$-env-endpoint Endpoint Overrides
+
```
[the remark this answers is not in the document]
Comment (coryan) [reply generator/internal/scaffold_generator.cc]: Removed, but I think it is the checkers script that adds the blank line.

File: generator/internal/scaffold_generator.cc [hunk 3]
```diff
@@ -676,6 +654,57 @@ can override the default policies.
...
+- `GOOGLE_CLOUD_CPP_TRACING_OPTIONS=...` modifies the behavior of gRPC tracing,
+  including whether messages will be output on multiple lines, or whether
+  string/bytes fields will be truncated.
+
+- `GOOGLE_CLOUD_CPP_ENABLE_CLOG=yes` turns on logging in the library. Basically
+  the library always "logs" but the logging infrastructure has no backend to
+  actually print anything until the application sets a backend or it sets this
+  environment variable.
+
+@see google::cloud::EndpointOption
```
[the remark this answers is not in the document]
Comment (coryan) [reply generator/internal/scaffold_generator.cc]: No. Removed and updated all the generated files.

File: generator/internal/scaffold_generator.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (devbww) [reply generator/internal/scaffold_generator.cc]: Ack.  I'll take a look later.

[the remark this answers is not in the document]
Comment (devbww) [reply generator/internal/scaffold_generator.cc]: Followup in #11387.

**Pull request merged**